### PR TITLE
ET-4553 Adding planning repo to GH role

### PR DESCRIPTION
**Pull request by jorge-cr-13**
## What 🕵️ 🔍

- What does the pull request add and/or solve?

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

**Review by Robert-Steiner — COMMENTED** [review]
what was the problem with the first solution?

**Comment by jorge-cr-13** [comment]
> > what was the problem with the first solution?
> 
> I guess because it was in the embeddings_service. then yeah it makes sense. could we reuse `gh_ecr_role.hcl` but with a different `role_name`? here is the module https://github.com/xaynetwork/xayn_terraform_modules/tree/main/generic/github_action_ecr_role

It can be reused, but it has this section:

`module "iam_github_oidc_provider" {
  source = "github.com/terraform-aws-modules/terraform-aws-iam//modules/iam-github-oidc-provider?ref=v5.8.0"

  tags = var.tags
}`

And this can't be repeated, so either we put this section out of the module or use the approach I used. What do you think is better?

**Comment by Robert-Steiner** [comment]
> > > what was the problem with the first solution?
> > 
> > 
> > I guess because it was in the embeddings_service. then yeah it makes sense. could we reuse `gh_ecr_role.hcl` but with a different `role_name`? here is the module https://github.com/xaynetwork/xayn_terraform_modules/tree/main/generic/github_action_ecr_role
> 
> It can be reused, but it has this section:
> 
> ```
>  module "iam_github_oidc_provider" {
>   source = "github.com/terraform-aws-modules/terraform-aws-iam//modules/iam-github-oidc-provider?ref=v5.8.0"
> 
>   tags = var.tags
> } 
> ```
> 
> And this can't be repeated, so either we put this section out of the module or use the approach I used. What do you think is better?

I see! then let's pull out the ocid provider module and create it directly in `organization/global/ocid`. then we can reuse the role module used for the roles

File: organization/global/oidc/terragrunt.hcl [hunk 1]
```diff
@@ -0,0 +1,12 @@
+locals {
+  account_vars = read_terragrunt_config(find_in_parent_folders("account.hcl"))
+  account_tags = local.account_vars.locals.tags
+}
+
```
Comment (Robert-Steiner) [inline organization/global/oidc/terragrunt.hcl:5]:
```suggestion
```

**Comment by Robert-Steiner** [comment]
needs a fmt, then we can merge it

**Review by Robert-Steiner — APPROVED** [review]

**Review by Robert-Steiner — APPROVED** [review]

**Pull request merged**